### ISSUE TITLE
Try using word2vec output trained over a large data set

### ISSUE THREAD

**Issue by ijkilchenko** [ijkilchenko/Fuzbal#3]
Current version, v0.91, uses the output from word2vec which is generated from their default data set (text file of 100 mbs). This output gives a json file which is about 150 mbs (and the extension is then compressed to about 55 mbs in the Chrome Store). This json file takes up about 400 mbs of RAM when the background page reads the file into a Javascript object (for instant vector lookup). 

For comparison, 400 mbs of RAM is 4x bigger than what Adblock uses in RAM and 8x bigger than what Lastpass uses. So reducing the word2vec output file somehow would be great. 

On the other hand, we need to experiment with training over large data sets and seeing how large the output files become. The added accuracy of word vectors might be very useful. 

Try some of the following things while keeping in mind the size/accuracy trade-offs. 
- Train over several gbs worth of training material
- Cut the number of dimensions in half (should save about half the space?)
- Round up the floating point values a few digits
- Somehow only keep most frequent words in the json format
- Split the json file into two files: one with 10% most frequent words and one with the other 90%
  - Allow the user a setting to choose to load the larger file into RAM as well

**Comment by ijkilchenko** [comment]
We just switched to [GloVe](http://nlp.stanford.edu/projects/glove/) away from [word2vec](https://code.google.com/p/word2vec/) for no other reason than that it was easier to find short pre-trained vectors from GloVe. These pretrained vectors are much, much more accurate than ones I have been working with. I am working 50 dimensional vectors (instead of previously 200 dimensional) and they seem more accurate (probably because they were trained over a larger data set). Finally, I found a text file with 20 k most frequent words in English and now keep only those in my `glove_small_dict.json`. All of this brought down the file size to 9 mbs and we're using about 50 mbs of RAM now (8x less).
